### PR TITLE
Feature 13 store content

### LINKED ISSUE #13: Store content
As a user I'd like the device to store whether content it is been assigned to

### PR DESCRIPTION
**Pull request by villejuutila**
Resolves #13

### CODE REVIEW

File: lib/screens/exhibition_screen.dart [hunk 1]
```diff
@@ -1,25 +1,67 @@
...
+
+  const ExhibitionScreen({Key? key, required this.exhibitionId})
+      : super(key: key);
 
   @override
   State<ExhibitionScreen> createState() => _ExhibitionScreenState();
 }
 
 /// Exhibition screen state
 class _ExhibitionScreenState extends State<ExhibitionScreen> {
```
Comment (anttileppa) [inline lib/screens/exhibition_screen.dart]: Rename ExhibitionScreen to PageScreen, this does not display an exhibition, it displaying single page in an Exhibition

File: lib/screens/exhibition_screen.dart [hunk 2]
```diff
@@ -1,25 +1,67 @@
...
-  late String _pageHtml;
+  String? _pageHtml;
+  final List<ExhibitionPageResource> _pageResources = [];
+  final List<ExhibitionPageEventTrigger> _eventTriggers = [];
+
+  /// Loads exhibition and its content by [exhibitionId]
+  Future _loadExhibition(String exhibitionId) async {
```
Comment (anttileppa) [inline lib/screens/exhibition_screen.dart]: _loadPage, change to page id

File: lib/mqtt/listeners/attach_listener.dart [hunk 3]
```diff
@@ -16,18 +24,58 @@ class AttachListener {
...
+    SimpleLogger().info("Handling attach message...");
+    final attachedMessage =
+        AbstractListener.decodeMessage<MqttDeviceAttachedToExhibition>(
+            message, MqttDeviceAttachedToExhibition);
+    SimpleLogger().info(attachedMessage);
+
+    // There's no support for having device attached to multiple exhibitions and therefore existing exhibitions are deleted
+    await pageDao.deletePages();
+    await layoutDao.deleteLayouts();
+    await exhibitionDao.deleteExhibitions();
```
Comment (anttileppa) [inline lib/mqtt/listeners/attach_listener.dart]: exhibition entity should be named more appropriately, it does not contain details from exhibition, it contains details how the device is connected to an exhibition, so more approperiate name would be e.g. DeviceExhibitionDetails

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: lib/screens/exhibition_screen.dart (hunk 1 above)
Comment (villejuutila) [reply lib/screens/exhibition_screen.dart]: Resolved.

File: lib/mqtt/listeners/attach_listener.dart (hunk 3 above)
Comment (villejuutila) [reply lib/mqtt/listeners/attach_listener.dart]: Resolved.

File: lib/screens/exhibition_screen.dart (hunk 2 above)
Comment (villejuutila) [reply lib/screens/exhibition_screen.dart]: Resolved.

**Review by anttileppa — APPROVED** [review]

**Pull request merged**